conan/tools/cmake/cmakedeps/templates/target_data.py

```diff
@@ -142,7 +142,14 @@ def get_dependency_filenames(self):
                     req = self.conanfile.dependencies.host_requires[dep_name]
                     ret.append(get_file_name(req))
         elif self.conanfile.dependencies.host_requires:
-            ret = [get_file_name(r) for r in self.conanfile.dependencies.host_requires]
+            # The consumer might be skipping some node that in the other hand is a valid require for
```

Comment:
This has been merged, but I don't think this is true. The "skipped" is a property of the node, not the require, it is impossible that one node is skipped for some packages and not others. The transitivity of "skipping" is correctly processed, if some node of the graph requires other node, even if the downstream bottom consumer doesn't need it, that dependency will not be skipped.

I have checked to remove this (and use the ``dependencies.non_skipped`` directly here, and it seems it works fine, and I have not been able to reproduce a failure here.